tlatools/org.lamport.tlatools/src/tlc2/overrides/TLCExt.java

```diff
@@ -0,0 +1,247 @@
+		} catch (EvalException | TLCRuntimeException e) {
+			final StringValue err = (StringValue) tool.eval(args[0], c, s0);
+			if (err.val.equals(e.getMessage())) {
+				return BoolValue.ValTrue;
+			}
+		}
+		return BoolValue.ValFalse;
+	}
+
+	private static final Scanner scanner = new Scanner(System.in);
+
+	// This is likely only useful with a single worker, but let's synchronize
+	// anyway.
+	@Evaluation(definition = "PickSuccessor", module = "TLCExt", warn = false)
+	public synchronized static Value pickSuccessor(final Tool tool, final ExprOrOpArgNode[] args, final Context c,
+			final TLCState s0, final TLCState s1, final int control, final CostModel cm) {
+
+		// TLC checks action constraints before it checks if states are new or not. Exclude seen states here
+		// to not repeatedly ask a user to extend a behavior with the same state over and over again.
+		try {
+			if (((ModelChecker) TLCGlobals.mainChecker).theFPSet.contains(s1.fingerPrint())) {
+				// If it is a seen state it is by definition in the model.
+				return BoolValue.ValTrue;
+			}
+		} catch (IOException notExpectedToHappen) {
+			notExpectedToHappen.printStackTrace();
+			return BoolValue.ValTrue;
+		}
+
+		// Eval expression and only let user interactively pick successor states when it
+		// evaluates to FALSE.
+		final Value guard = tool.eval(args[0], c, s0, s1, control, cm);
+		if (!(guard instanceof BoolValue)) {
+			Assert.fail("In evaluating TLCExt!PickSuccessor, a non-boolean expression (" + guard.getKindString()
+					+ ") was used as the condition " + "of an IF.\n" + args[0]);
+		}
+		if (((BoolValue) guard).val) {
+			return BoolValue.ValTrue;
+		}
+
+		// Find the (first) Action this pair of states belongs to. If more than one
+		// Action match, we pick the first one.
+		// TODO: This is clumsy (we regenerate all next-states again) and incorrect if
+		// two actions generate the same successor states. It's good enough for now
+		// until the Action instance was passed down the call-stack.
+		Action action = null;
+		LOOP: for (Action act : tool.getActions()) {
+			StateVec nextStates = tool.getNextStates(act, s0);
+			if (nextStates.contains(s1)) {
+				action = act;
+				break LOOP;
+			}
+		}
+
+		while (true) {
+			// TODO Add more commands such as continue/resume to pick every successor,
+			// randomly pick next N successors, terminate to stop state space exploration,
+			// ...
+			MP.printMessage(EC.TLC_MODULE_OVERRIDE_STDOUT,
+					String.format(
+							"Extend behavior of length %s with a \"%s\" step [%s]? (Yes/no/explored/states/diff):",
+							s0.getLevel(), action.getName(), action));
+
+			final String nextLine = scanner.nextLine();
+			if (nextLine.trim().isEmpty() || nextLine.toLowerCase().startsWith("y")) {
+				return BoolValue.ValTrue;
+			} else if (nextLine.charAt(0) == 's') {
+				MP.printMessage(EC.TLC_MODULE_OVERRIDE_STDOUT,
+						String.format("%s\n~>\n%s", s0.toString().trim(), s1.toString().trim()));
```

Comment:
Our static analysis engine detects that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.

Reply:
@aimiebell Thanks, please provide a pull request.